insights/parsers/softnet_stat.py

```diff
@@ -0,0 +1,99 @@
+performed time_squeeze(third column), cpu collision i.e collision occur while obtaining
+device lock while transmitting cpu_collision packets (Ninth column), received_rps
+number of times cpu woken up received_rps (tenth column), number of times reached flow
+limit count flow_limit_count (eleventh column).
+
+    The typical contents of ``cat /proc/net/softnet_stat`` file is as below::
+
+        00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+
+        Column-01: packet_process: Packet processed by each CPU.
+        Column-02: packet_drop: Packets droped.
+        Column-03: time_squeeze: net_rx_action.
+        Column-09: cpu_collision: collision occur while obtaining device lock while transmitting.
+        Column-10: received_rps: number of times cpu woken up received_rps.
+        Column-11: flow_limit_count: number of times reached flow limit count.
+
+    Examples:
+        >>> SOFTNET_STATS = '''
+        ... 00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ...'''.strip()
+        >>> softnet_obj = context_wrap(SOFTNET_STATS)
+        >>> softnet_obj.is_packet_drops()
+        True
+        >>> softnet_obj.cpu_instances
+        4
+        >>> softnet_obj.per_cpu_nstat('packet_drops')
+        [0, 0, 0, 1]
+"""
+
+from .. import Parser, parser
+
+
+@parser("softnet_stat")
+class SoftNetStats(Parser):
+    """Parses ``/proc/net/softnet_stat`` file contains"""
+
+    def __init__(self, context):
+        self.cpu_instances = None
+        self.data = {}
+        super(SoftNetStats, self).__init__(context)
+
+    def parse_content(self, contents):
+        cpu = 0
+        nstats = {}
+        for line in contents:
+            stats = line.split(None)
+
+            nstats[cpu] = {'packet_process': stats[0],
+                            'packet_drops': stats[1],
+                            'time_squeeze': stats[2],
+                            'cpu_collision': stats[8],
+                            'received_rps': stats[9],
+                            'flow_limit_count': stats[10]}
+            cpu = cpu + 1
+        self.cpu_instances = cpu
+        self.data = nstats
+
+    def per_cpu_nstat(self, key):
```

Comment:
In addition to this interface, do you think it would be better to add another one such as `nstat_per_cpu[0]`.

Using `list` instead of `dict` to store the `nstats` with the meaningful name can solve this.

Reply:
yes its is important because we are going to return a column let's say packet_drops for all cpu , so "key" will be packet_drops , packet_drops  per cpu